client/transport.c

```diff
@@ -298,11 +346,25 @@ static int do_request(int type, xdag_time_t start_time, xdag_time_t end_time, vo
 		dnet_send_xdag_packet(&b, reply_connection);
 	}
 
-	for (t = time(0); reply_result < 0 && time(0) - t < REQUEST_WAIT; ) {
-		sleep(1);
+	time(&actual_time);
+	expire_time.tv_sec = actual_time + REQUEST_WAIT;
+
+	while(!reply_rcvd){
```

Comment:
When the previous connection is down then process_transport_block will not be called, and reply_rcvd will always be 0, here will be in while(true) loop no request anymore.

Reply:
pthread_cond_timedwait(&g_process_cond, &g_process_mutex, &expire_time) will return -1 when wait time will expire and so it will exit from the loop